numpyro/distributions/directional.py

```diff
@@ -75,6 +132,246 @@ def variance(self):
         )
 
 
+PhiMarginalState = namedtuple("PhiMarginalState", ["i", "done", "phi", "key"])
+
+
+class SineBivariateVonMises(Distribution):
+    r"""Unimodal distribution of two dependent angles on the 2-torus (S^1 ⨂ S^1) given by
+
+    .. math::
+        C^{-1}\exp(\kappa_1\cos(x_1-\mu_1) + \kappa_2\cos(x_2 -\mu_2) + \rho\sin(x_1 - \mu_1)\sin(x_2 - \mu_2))
+
+    and
+
+    .. math::
+        C = (2\pi)^2 \sum_{i=0} {2i \choose i}
+        \left(\frac{\rho^2}{4\kappa_1\kappa_2}\right)^i I_i(\kappa_1)I_i(\kappa_2),
+
+    where :math:`I_i(\cdot)` is the modified bessel function of first kind, mu's are the locations of the distribution,
+    kappa's are the concentration and rho gives the correlation between angles :math:`x_1` and :math:`x_2`.
+    This distribution is helpful for modeling coupled angles such as torsion angles in peptide chains.
+
+    To infer parameters, use :class:`~numpyro.infer.NUTS` or :class:`~numpyro.infer.HMC` with priors that
+    avoid parameterizations where the distribution becomes bimodal; see note below.
+
+    .. note:: Sample efficiency drops as
+
+        .. math::
+            \frac{\rho}{\kappa_1\kappa_2} \rightarrow 1
+
+        because the distribution becomes increasingly bimodal.
+
+    .. note:: The correlation and weighted_correlation params are mutually exclusive.
+
+    .. note:: In the context of :class:`~numpyro.infer.SVI`, this distribution can be used as a likelihood but not for
+        latent variables.
+
+    ** References: **
+        1. Probabilistic model for two dependent circular variables Singh, H., Hnizdo, V., and Demchuck, E. (2002)
+
+    :param jnp.Tensor phi_loc: location of first angle
+    :param jnp.Tensor psi_loc: location of second angle
+    :param jnp.Tensor phi_concentration: concentration of first angle
+    :param jnp.Tensor psi_concentration: concentration of second angle
+    :param jnp.Tensor correlation: correlation between the two angles
+    :param jnp.Tensor weighted_correlation: set correlation to weigthed_corr * sqrt(phi_conc*psi_conc)
+        to avoid bimodality (see note).
+    """
+
+    arg_constraints = {
+        "phi_loc": constraints.real,
+        "psi_loc": constraints.real,
+        "phi_concentration": constraints.positive,
+        "psi_concentration": constraints.positive,
+        "correlation": constraints.real,
+    }
+    support = constraints.independent(constraints.real, 1)
```

Comment:
I'm not sure if we need to be consistent with VonMises here. Could you add # TODO at those supports to revisit the issue?

Reply:
I believe we do, I'll add a TODO and revisit once #1080 is settled.